src/main/scala/workflow/graph/PipelineDatasetOut.scala

```diff
@@ -0,0 +1,23 @@
+package workflow.graph
+
+import org.apache.spark.rdd.RDD
+
+/**
+ * This class is a lazy wrapper around the output of a pipeline that was passed an RDD as input.
+ *
+ * Under the hood, it extends [[PipelineResult]] and keeps track of the necessary execution plan.
+ */
+class PipelineDatasetOut[T] private[graph] (executor: GraphExecutor, sink: SinkId)
+  extends PipelineResult[RDD[T]](
+    executor,
+    sink)
+
+object PipelineDatasetOut {
```

Comment:
If this is going to be user / developer visible I like `PipelineDataset` better. If not it matters less